commons/Logger.ts

```diff
@@ -1,54 +1,65 @@
 let logId = 0;
 class Log implements ILog {
-  public level: string = process.env.DEBUG ? 'stats' : 'warn';
+  public readonly level: string = process.env.DEBUG ? 'stats' : 'warn';
   private readonly module: string;
+  private readonly logLevel: number;
+
   constructor(module: NodeModule) {
+    this.logLevel = logLevels.indexOf(this.level);
     this.module = module ? extractPathFromModule(module) : '';
   }
 
-  public stats(action: string, data?: any, parentMessageId?: number) {
-    if (this.level === 'stats') {
-      console.log(
-        `${new Date().toISOString()} STATS ${action}`,
-        ...[parentMessageId, data].filter(x => x !== undefined && x !== null),
-      );
-      return (logId += 1);
-    }
+  public stats(sessionId: string, action: string, data?: any, parentLogId?: number) {
+    return this.log('stats', sessionId, action, data, parentLogId);
   }
 
-  public info(action: string, data?: any, parentMessageId?: number) {
-    if (this.level === 'stats' || this.level === 'info') {
-      console.log(
-        `${new Date().toISOString()} INFO ${action}`,
-        ...[parentMessageId, data].filter(x => x !== undefined && x !== null),
-      );
-      return (logId += 1);
-    }
+  public info(sessionId: string, action: string, data?: any, parentLogId?: number) {
+    return this.log('info', sessionId, action, data, parentLogId);
   }
```

Comment:
It seems these args types open the door for mistakes. For example, `log.stats(action, data)` is valid so long as data is a string. I believe this log item would then drop through the cracks since `action` matches no valid `sessionId`.

Perhaps this isn't much of a concern.

Regardless, I'm not sure of a legit way to fix this other than switching sessionId to be a `number`, which would be a broad undertaking.

Reply:
It's a good point. I just tried explicitly making data an object. This helps a small amount, and found a few log problems that were in the current pr. I don't think it makes it impossible to send in a bad log though. I think I would need to move more towards sessionId being an optional param off of data for that to be the case. Or maybe if you send data, it has to include a sessionId if you send it (so you'd have to explicitly say null)

Reply:
Based on this, I decided to change the logging signature to have action, then a data object that, if passed, requires you to say what sessionId the logs are associated with.